standard/basic-concepts.md

```diff
@@ -10,33 +10,45 @@ A program compiled as an application shall contain at least one method qualifyin
 
 > *Note*: On systems supporting a command line, application parameters correspond to what are generally known as command-line arguments. *end note*
 
-If the entry point’s return type is `int` rather than `void`, the return value from the method invocation by the execution environment is used in application termination ([§7.2](basic-concepts.md#72-application-termination)).
+If the entry point's return type is `int` or `System.Threading.Tasks.Task<int>`, the return value from the method invocation by the execution environment is used in application termination ([§7.2](basic-concepts.md#72-application-termination)).
 
 Other than the situations listed above, entry point methods behave like those that are not entry points in every respect. In particular, if the entry point is invoked at any other point during the application’s lifetime, such as by regular method invocation, there is no special handling of the method: if there is a parameter, it may have an initial value of `null`, or a non-`null` value referring to an array that contains null references. Likewise, the return value of the entry point has no special significance other than in the invocation from the execution environment.
 
 ## 7.2 Application termination
 
 ***Application termination*** returns control to the execution environment.
 
-If the return type of the application’s entry point method is `int`, the value returned serves as the application’s ***termination status code***. The purpose of this code is to allow communication of success or failure to the execution environment.
+If the return type of the application's entry point method is `int` or `System.Threading.Tasks.Task<int>`, the value of the `int` returned serves as the application's ***termination status code***. The purpose of this code is to allow communication of success or failure to the execution environment.
 
-If the return type of the entry point method is `void`, reaching the right brace (`}`) that terminates that method, or executing a `return` statement that has no expression, results in a termination status code of `0`. If the entry point method terminates due to an exception ([§20.4](exceptions.md#204-how-exceptions-are-handled)), the exit code is implementation-specific. Additionally, the implementation may provide alternative APIs for specifying the exit code.
+If the return type of the entry point method is `void` or `System.Threading.Tasks.Task`, reaching the right brace (`}`) that terminates that method, or executing a `return` statement that has no expression, results in a termination status code of `0`. If the entry point method terminates due to an exception ([§20.4](exceptions.md#204-how-exceptions-are-handled)), the exit code is implementation-specific. Additionally, the implementation may provide alternative APIs for specifying the exit code.
```

Comment:
OK, it does not matter what this paragraph says about a method whose return type is `System.Threading.Tasks.Task` but which does not have the `async` modifier. If the final right brace of such a method looks reachable to the compiler, or the method contains `return;`, then it cannot even be compiled, so the runtime behavior does not matter.

Reply:
I'd argue that's covered elsewhere, and that this paragraph describes the behavior of code that *does* compile.

However, I do wonder whether we can just simplify it to say that if the return type is `void` and the method completes without an exception, or if the return type is `Task` and the returned task completes successfully (we can wordsmith that to be consistent with async elsewhere) then the termination status code is 0.

(It's actually not clear to me whether the termination status code is always 0 if the Main method completes but there are other non-daemon threads - one of those could call Environment.Exit with a non-zero status code for example.)

Reply:
Also it might be expression-bodied, therefore not have a `}`

Reply:
I've avoided these issues by a) introducing the concept of an effective entry point (which is always synchronous); b) talking about execution completing without an exception rather than talking about reaching a `}`.